libbeat/api/server.go

```diff
@@ -72,6 +72,25 @@ func (s *Server) Stop() error {
 	return s.l.Close()
 }
 
+// AttachHandler will attach a handler at the specified route and return an error instead of panicing.
+func (s *Server) AttachHandler(route string, h http.Handler) (err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			switch r.(type) {
+			case error:
+				err = r.(error)
+			case string:
+				err = fmt.Errorf(r.(string))
+			default:
+				err = fmt.Errorf("handle attempted to panic with %v", r)
+			}
```

Comment:
```suggestion
			switch r := r.(type) {
			case error:
				err = r
			case string:
				err = errors.New(r)
			default:
				err = fmt.Errorf("handle attempted to panic with %v", r)
			}
```

Needs "errors" added to imports.